cmd/clusterctl/client/common.go

```diff
@@ -62,7 +62,7 @@ func (c *clusterctlClient) getComponentsByName(ctx context.Context, provider str
 
 // parseProviderName defines a utility function that parses the abbreviated syntax for name[:version].
 func parseProviderName(provider string) (name string, version string, err error) {
-	t := strings.Split(strings.ToLower(provider), ":")
```

Comment:
Do we know why this was done?

Reply:
Please let's keep to lower on the provider name at least (AFAIK this was added to be tolerate user typos); since we are at it, wdyt abouy enforcing lower case names for provider to ensure consistency

e.g. in https://github.com/kubernetes-sigs/cluster-api/blob/f83ac54d53b46fc551e5550809ed2e898ee8ee55/cmd/clusterctl/client/config/providers_client.go#L524

+ documenting this into

https://cluster-api.sigs.k8s.io/developer/providers/contracts/clusterctl#adding-a-provider-to-clusterctl

(and the 1.11 migration doc, we can make this a v1beta2 contract thing)

Reply:
If I'm not wrong we are missing the test for lower case name in validateProvider